internal/security/bootstrapper/tcp/client_test.go

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package tcp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDialTcpClient(t *testing.T) {
+	lc := logger.MockLogger{}
+	errs := make(chan error, 1)
+	testListeningPort := 12333
+	srv := NewTcpServer()
+	go func() {
+		errs <- srv.StartListener(testListeningPort, lc, "")
+	}()
+
+	time.Sleep(time.Second)
+
+	tests := []struct {
+		name string
+		host string
+		port int
+	}{
+		{"Ignore empty host input", "", testListeningPort},
+		{"Ignore port number 0", "localhost", 0},
+		{"Ignore both empty host and 0 port number", "", 0},
+		{"Dial the TCP server and port", "localhost", testListeningPort},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// dialTcp in goroutine so it won't block forever
+			go func() {
+				errs <- DialTcp(tt.host, tt.port, lc)
+			}()
+
+			// give some time for connection
+			time.Sleep(2 * time.Second)
+
+			require.NoError(t, <-errs)
```

Comment:
This could block for ever. So the go func just delays it...
put in a select with a timeout case:
```
	select {
	case err <-errs:
		require.NoError(t, <-errs)
	case <-time.After(2 * time.Second):
		require.Fail(t, "DialTcp never returned")
	}

Reply:
ok